### ISSUE TITLE
No differentiation between completed lessons and locked lessons in topics tab

### ISSUE THREAD

**Issue by vrajdesai78** [oppia/oppia-android#4648]
**Describe the bug**
In topic lessons tab, when we use screen reader there is not differentiation between completed lessons and locked lessons. Also, locked lessons are having content description as "lessons in progress" which should be locked lessons. 

**To Reproduce**
1. Go to Any Story
2. Click on Lessons Tab
3. Turn on talkback reader
4. Navigate to the list of lessons. 

**Expected behavior**
The content description for completed lessons, locked lessons and In Progress lesson should be appropriate to it's state. For example, for locked chapter content description should have locked chapter in the end. 

**Demonstrationtion**
https://user-images.githubusercontent.com/43074241/195417529-357236fe-1d37-45a1-888f-51158ea89822.mp4

**Environment**
- Device/emulator being used: Realme GT Master Edition
- Android or SDK version (e.g. Android 5 or SDK 21): Android 12
- App version (you can get this through system app settings or via the admin controls menu in-app): 1.0

**Comment by JishnuGoyal** [comment]
Hi @vrajdesai78! Nice find! Although the TalkBack should have automatically worked as expected correctly, as there weren't any specific changes introduced to that part, but I'll investigate on this and work on a fix.

Thanks!

**Comment by JishnuGoyal** [comment]
Unassigning myself so that someone else can pick this issue. Although I haven't digged into the details but this might be a good first/second issue. WDYT @BenHenning ?

**Comment by BenHenning** [comment]
I think so, but @JishnuGoyal could you provide a few code points and a brief write-up (e.g. bullet items of what roughly needs to change; this should be in a reply comment not a doc) of how you think this can be solved? That's more or less the bar that we're trying to establish for starter issues moving forward (to reduce the back-and-forths needed with new contributors who are trying to familiarize themselves).

**Comment by kkmurerwa** [comment]
### Problem Statement and Cause
I was able to successfully replicate the issue. As stated in the issue, In-progress and completed lessons are read correctly but locked lessons are considered in-progress lessons. 

This is caused by a logic flaw. The control structure in the `computeChapterPlayStateIconContentDescription()` of the [ChapterSummaryViewModel](https://github.com/kkmurerwa/oppia-android/blob/1ba98ddb66590e7879ec01928ac6e59cd9da4408/app/src/main/java/org/oppia/android/app/topic/lessons/ChapterSummaryViewModel.kt) assumes that any `ChapterPlayState` that is not `ChapterPlayState.COMPLETED` is automatically a `ChapterPlayState.IN_PROGRESS_*`. This is what makes Talkback to read the content description to locked lessons as in-progress lessons.

### Proposed Solution
Below is my suggested approach to the problem;
- Update the `ChapterSummaryViewModel` and add a string constructor parameter `val previousChapterTitle: String`. This will be needed to compile the content description needed for Talkback in the locked lessons.
- Update the [StorySummaryViewModel](https://github.com/kkmurerwa/oppia-android/blob/1ba98ddb66590e7879ec01928ac6e59cd9da4408/app/src/main/java/org/oppia/android/app/topic/lessons/StorySummaryViewModel.kt) on the `computeChapterSummaryItemList()` method and add an argument for the `previousChapterTitle` parameter.
- Add a condition in the if statement on the `computeChapterPlayStateIconContentDescription()` method for the `ChapterPlayState.NOT_PLAYABLE_MISSING_PREREQUISITES` state. This will differentiate locked lessons from in-progress lessons, since they typically have the missing prerequisites state.
- In the `ChapterPlayState.NOT_PLAYABLE_MISSING_PREREQUISITES` statement body, invoke the `resourceHandler.getStringInLocaleWithWrapping()` with the `R.string.chapter_prerequisite_title_label` string already existing as a string resource.
- Update the call to also include `index` instead of `index + 1`. Because we need to pass the previous chapter title to the talkback. We will also add our `previousChapterTitle` variable to the method call. Our final condition should look like this;
    
    ```kotlin
    ChapterPlayState.NOT_PLAYABLE_MISSING_PREREQUISITES -> {
        resourceHandler.getStringInLocaleWithWrapping(
          R.string.chapter_prerequisite_title_label, (index).toString(), previousChapterTitle
        )
    }

That should be able to successfully fix the issue.
